presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraMetadata.java

```diff
@@ -219,17 +219,20 @@ public boolean usesLegacyTableLayouts()
             unenforcedConstraint = clusteringPredicatesExtractor.getUnenforcedConstraints();
         }
 
-        if (handle.getPartitions().containsAll(partitionResult.getPartitions()) && handle.getClusteringKeyPredicates().equals(clusteringKeyPredicates)) {
+        Optional<List<CassandraPartition>> currentPartitions = handle.getPartitions();
+        if (currentPartitions.isPresent() &&
+                currentPartitions.get().containsAll(partitionResult.getPartitions()) &&
+                handle.getClusteringKeyPredicates().equals(clusteringKeyPredicates)) {
             return Optional.empty();
         }
 
         return Optional.of(
                 new ConstraintApplicationResult<>(new CassandraTableHandle(
                         handle.getSchemaName(),
                         handle.getTableName(),
-                        partitionResult.getPartitions(),
+                        Optional.of(partitionResult.getPartitions()),
```

Comment:
Should this be an intersection of `partitionResult.getPartitions()` and `currentPartitions` (if present)?
In `io.prestosql.plugin.cassandra.CassandraPartitionManager#getPartitions` i don't see how yet how subsequent call to applyFilter should not be able to _widen_ (instead of _narrowing_) the list of partitions we're querying.

@ebyhr can you double check me on Cassandra changes